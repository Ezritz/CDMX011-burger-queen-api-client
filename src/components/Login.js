import styles from '../css/Login.scss';
import React,{useState} from 'react';
import { useLogin } from '../hooks/useLogin';
import myImage from '../images/Logo.png';

// import {useState} from 'react';

export default function Login() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const { logIn, error, isPending } = useLogin();
  
  const handleSubmit = (e) => {
    e.preventDefault();
    logIn(email,password);
  }
  

  return(
    <div id='container'>
      <div id='container-img'>

        <img src={myImage} alt='logo BQ' className='imgLogo'/>
      </div>
      <div id='container-form' className= {styles['']}>
        <form onSubmit={handleSubmit}>

          <label className = 'label-1' htmlFor="inputEmail">
            <span>E-mail:</span>
            
          </label>
          <br/>
          <input 
            className='input-1'
            type="email"
            onChange={(e) => setEmail(e.target.value)}
            value={email}
            id="inputEmail">
          </input>
          <br/>
          <label htmlFor="inputPassword" className='label-2'>
            <span>Contrase√±a:</span>
            
          </label>
          <br/>
          <input 
            className='input-2'
            type="password"
            onChange={(e) => setPassword(e.target.value)}
            value={password}
            id="inputPassword">
          </input>
          {error && <p>{error}</p>}
          <p></p>
          <br/>
          {!isPending && <button 
          className='btn'
          >Iniciar Sesion</button>}
          {isPending && <button className='btn' disabled>Cargando</button>}

        </form>
      </div>
      
    </div>
  )
}
