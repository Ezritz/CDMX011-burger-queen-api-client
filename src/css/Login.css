@import url("https://fonts.googleapis.com/css2?family=Hind+Madurai&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Rum+Raisin&display=swap");
body {
  background-color: #DDB892;
}

#container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-wrap: nowrap;
      flex-wrap: nowrap;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
  -ms-flex-pack: distribute;
      justify-content: space-around;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  margin: 10% 5%;
}

img {
  width: 400px;
}

.input-1 {
  border-radius: 10px;
  background-color: #E6CCB2;
  border: 1px solid #8D563A;
  margin: 4px auto 40px;
  font-family: 'Hind Madurai', sans-serif;
  text-align: center;
  width: 80%;
  height: 3em;
}

.input-2 {
  border-radius: 10px;
  background-color: #E6CCB2;
  border: 1px solid #8D563A;
  margin: 4px auto 40px;
  font-family: 'Hind Madurai', sans-serif;
  text-align: center;
  width: 80%;
  height: 3em;
}

.label-1 {
  font-family: 'Rum Raisin', sans-serif;
  font-size: 36px;
  margin: 0;
  padding-left: 16%;
}

.label-2 {
  font-family: 'Rum Raisin', sans-serif;
  font-size: 36px;
  margin: 0;
  padding-left: 16%;
}

form {
  border: 1px solid #650002;
  width: 468px;
  height: 536px;
  border-radius: 50px;
  padding: auto;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: left;
      -ms-flex-align: left;
          align-items: left;
}

.btn {
  border-radius: 10px;
  width: 60%;
  height: 2em;
  background-color: #7A3F25;
  border: none;
  margin-left: 20%;
  margin-right: 20%;
  font-family: 'Rum Raisin', sans-serif;
  font-size: 28px;
  color: #E6CCB2;
}
/*# sourceMappingURL=Login.css.map */