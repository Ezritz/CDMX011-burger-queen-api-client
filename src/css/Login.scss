@import url('https://fonts.googleapis.com/css2?family=Hind+Madurai&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Rum+Raisin&display=swap');

$color-background: #DDB892;
$color-bc-inputs: #E6CCB2;
$color-border-inputs: #8D563A;
$color-bc-btn: #7A3F25;
$color-border-form: #650002;

body{
    background-color: $color-background;
}
#container {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    margin: 10% 5%;
}

img {
    width: 400px;
}

@for $i from 1 through 2 {
    .input-#{$i} 
    {
        border-radius: 10px;
        background-color: $color-bc-inputs;
        border: 1px solid $color-border-inputs;
        margin: 4px auto 40px;
        font-family: 'Hind Madurai', sans-serif;
        text-align: center;
        width: 80%;
        height: 3em;
       

    }
}

@for $i from 1 through 2 {
    .label-#{$i} 
    {
        font-family: 'Rum Raisin', sans-serif;
        font-size: 36px;
        margin: 0;
        padding-left: 16%;
        
    }
}

form {
    border: 1px solid $color-border-form;
    width: 468px;
    height: 536px;
    border-radius: 50px;
    padding: auto;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: left;
}

.btn {
    border-radius: 10px;
    width: 60%;
    height: 2em;
    background-color: $color-bc-btn;
    border:none;
    margin-left: 20%;
    margin-right: 20%;
    font-family: 'Rum Raisin', sans-serif;
    font-size: 28px;
    color: $color-bc-inputs;
    
}

p{
    font-family: 'Hind Madurai', sans-serif;
    font-size: 15px;
    text-align: center;
}



