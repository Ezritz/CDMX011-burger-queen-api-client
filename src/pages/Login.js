import styles from './Login.css';
import React,{useState} from 'react';
import auth from '../firebase/config';
import {useNavigate} from 'react-router-dom';
// import {useState} from 'react';

const Login = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const navigate = useNavigate();

  const loginUser = (e) => {
    e.preventDefault();
    auth.signInWithEmailAndPassword(email,password)
      .then(() => {
        navigate('/home');
        console.log('Login exitoso')
      })
      .catch((err) => console.log('error', err.message))
  }
  
  return(
    <div>
      <form onSubmit={loginUser}className={styles['login-form']}>
        <label htmlFor="">
          <span>E-mail:</span>
          <input 
            type="email"
            onChange={(e) => setEmail(e.target.value)}
            id="inputEmail">
          </input>
        </label>
        <label htmlFor="">
          <span>Contrase√±a:</span>
          <input type="password"
            onChange={(e) => setPassword(e.target.value)}
            id="inputPassword">
          </input>
        </label>
        <button 
        className='btn'
        
        >Iniciar Sesion</button>

      </form>
    </div>
  )
}

export default Login;