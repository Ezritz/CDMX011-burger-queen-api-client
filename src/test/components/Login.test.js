import { render, screen, fireEvent, findAllByTestId, getByLabelText} from '@testing-library/react';
import React from 'react';
import Login from '../../components/Login';
import {MemoryRouter} from 'react-router-dom';

beforeEach(() => render(<Login/>))
describe("login ", () => {
  test('should render', () => {
    expect(render(<MemoryRouter><Login></Login></MemoryRouter>)).toMatchSnapshot();
  });
  test("Form", ()=> {
    const mockLogIn = jest.fn();

    const email = screen.getByLabelText('E-mail:');
    const password = screen.getByLabelText('ContraseÃ±a:');
    const btn = screen.getByRole('button',{name:/Iniciar Sesion/i});

    expect(email).toBeInTheDocument();
    expect(password).toBeInTheDocument();
    expect(btn).toBeInTheDocument();

    const emailValue = 'ejemplo@correo.com';
    const passValue = '123456';

    fireEvent.change(email , {target:{value: emailValue}});
    fireEvent.change(password , {target:{value: passValue}});
    fireEvent.click(btn);

    expect(mockLogIn).toHaveBeenCalledWith(emailValue, passValue);
  })
});

/*
describe('Login', () => {
    test('with valid inputs', () => {
        it.todo('calls the onSubmit function', () => {
            const mockOnSubmit = jest.fn();
            const{getByLabelText,getByRole} = render(<Login onSubmit={mockOnSubmit}/>)


            fireEvent.change(getByLabelText('E-mail:'),{target:{value:"email@test.com"}})
        })
    })
})
*/