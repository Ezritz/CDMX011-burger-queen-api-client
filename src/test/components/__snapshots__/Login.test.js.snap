// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login  should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="container"
      >
        <div
          id="container-img"
        >
          <img
            alt="logo BQ"
            src="Logo.png"
          />
        </div>
        <div
          id="container-form"
        >
          <form>
            <label
              class="label-1"
              for="inputEmail"
            >
              <span>
                E-mail:
              </span>
            </label>
            <br />
            <input
              class="input-1"
              id="inputEmail"
              type="email"
            />
            <br />
            <label
              class="label-2"
              for="inputPassword"
            >
              <span>
                Contraseña:
              </span>
            </label>
            <br />
            <input
              class="input-2"
              id="inputPassword"
              type="password"
            />
            <br />
            <button
              class="btn"
            >
              Iniciar Sesion
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="container"
    >
      <div
        id="container-img"
      >
        <img
          alt="logo BQ"
          src="Logo.png"
        />
      </div>
      <div
        id="container-form"
      >
        <form>
          <label
            class="label-1"
            for="inputEmail"
          >
            <span>
              E-mail:
            </span>
          </label>
          <br />
          <input
            class="input-1"
            id="inputEmail"
            type="email"
          />
          <br />
          <label
            class="label-2"
            for="inputPassword"
          >
            <span>
              Contraseña:
            </span>
          </label>
          <br />
          <input
            class="input-2"
            id="inputPassword"
            type="password"
          />
          <br />
          <button
            class="btn"
          >
            Iniciar Sesion
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
